### PR TITLE
Fix mech search by using GCP custom search

### PR DESCRIPTION
**Pull request by angrybayblade**

### CODE REVIEW

**Review by 0xArdi — APPROVED** [review]

File: tools/prediction_request.py [hunk 1]
```diff
@@ -112,11 +112,30 @@
...
+def get_urls_from_queries(queries: List[str], api_key: str, engine: str) -> List[str]:
     """Get URLs from search engine queries"""
     results = []
     for query in queries:
-        for url in googlesearch.search(query, num_results=3):
+        for url in search_google(
+            query=query,
+            api_key=api_key,
+            engine=engine,
+            num=3,
```
Comment (jmoreira-valory) [inline tools/prediction_request.py:137]:
```suggestion
            query=query,
            api_key=api_key,
            engine=engine,
            num=3, # Number of returned results
```

Comment (jmoreira-valory) [reply tools/prediction_request.py:137]: Also, why specify num=3 here if it's already the default value of the called method?

**Review by jmoreira-valory — CHANGES_REQUESTED** [review]
See comments.

File: tools/prediction_request.py [hunk 2]
```diff
@@ -207,7 +232,12 @@ def run(**kwargs) -> Tuple[str, Optional[Dict[str, Any]]]:
     engine = TOOL_TO_ENGINE[tool]
     additional_information = (
         fetch_additional_information(
-            prompt=prompt, engine=engine, temperature=temperature, max_tokens=max_tokens
+            prompt=prompt,
+            engine=engine,
+            temperature=temperature,
+            max_tokens=max_tokens,
+            google_api_key=kwargs["api_keys"]["google_api_key"],
+            google_engine=kwargs["api_keys"]["google_engine_id"],
```
[the remark this answers is not in the document]
Comment (angrybayblade) [reply tools/prediction_request.py:240]: I don't think we have time to define a whole new param ATM. We can improve on this later.

[the remark this answers is not in the document]
Comment (jmoreira-valory) [reply tools/prediction_request.py:240]: Please hardcode the value using a constant OR open an issue OR add a # FIXME comment.
These kind of quick solutions later create a lot of confusion for users.

[the remark this answers is not in the document]
Comment (angrybayblade) [reply tools/prediction_request.py:240]: Cannot hardcode it since it's provided by GCP and it changes with every user

**Review by jmoreira-valory — APPROVED** [review]
Please open an issue to address the comment about the engine later.

**Pull request merged**